GitUI/CommandsDialogs/FormCommit.cs

```diff
@@ -1185,11 +1187,13 @@ private void ShowChanges(GitItemStatus item, bool staged)
                 return;
             }
 
-            ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+            var task = ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
             {
                 await SetSelectedDiffAsync(item, staged);
                 _selectedDiffReloaded = true;
-            }).FileAndForget();
+            });
+            _joinableTasks.Add(task);
+            task.FileAndForget();
```

Comment:
You can make this step implicit by using a jtf instance that is created specifically for your `_joinableTasks` collection, so that the tasks are added automatically. 
Just add this to your ctor:
```cs
_joinableTaskFactory = ThreadHelper.JoinableTaskContext.CreateFactory(_joinableTasks);
```
Then use that jtf instead of ThreadHelper.JTF here (and anywhere else you call RunAsync without awaiting the result).

Reply:
We still want to be able to join outstanding operations started by `ThreadHelper.JoinableTaskFactory`. Is there a way to automatically add these fire-and-forget operations to both collections automatically?

Reply:
What are "both collections" that you refer to? There's only the one collection that's created in this class. `ThreadHelper.JoinableTaskFactory` has no collection backing it.

Reply:
Our `ThreadHelper` provides a collection to the factory:
https://github.com/gitextensions/gitextensions/blob/a2a6defafb96de0847ff92bfe01cb8275fa55084/GitExtUtils/GitUI/ThreadHelper.cs#L45-L47

Reply:
Oh! I have a hard time *not* reading `ThreadHelper` to mean the one in VS, which doesn't have a backing collection. 
No, there's no way at present to have one instance of JTF add a JoinableTask to *two* collections automatically.